File: app/src/main/java/ValidateUser/UserManager.java
package ValidateUser;


import android.content.Context;
import android.widget.EditText;

public class UserManager extends User{

    public UserManager(Context context,  EditText email,  EditText password, EditText confirmPassword) {
        super( email, password, confirmPassword);
        super.setContext(context);
    }

    public UserManager(Context context,EditText email, EditText password){
      super(email, password);
      super.setContext(context);
    }

    public boolean UserInputRequirementsFailedAtSignUp() {
        return  validateEmailFailed() || validatePasswordFailed() || validateConfirmPasswordFailed();
    }
    public boolean UserInputRequirementsFailedAtSignIn(){
        return validateEmailFailed() || validatePasswordFailed();
    }

}

File: app/src/main/java/ValidateUser/UserManager.kt
package ValidateUser

import android.content.Context
import android.widget.EditText

class UserManager(
    context: Context,
    email: EditText,
    password: EditText,
    confirmPassword: EditText?
) : User(context, email, password, confirmPassword) {


    fun userInputRequirementsFailedAtSignUp(): Boolean {
        return super.validateEmailFailed() || super.validatePasswordFailed() || super.validateConfirmPasswordFailed()
    }

    fun userInputRequirementsFailedAtSignIn(): Boolean {
        return super.validateEmailFailed() || super.validatePasswordFailed()
    }
}